### PR TITLE
test/e2e: add user workload monitoring

### PR DESCRIPTION
**Pull request by s-urbaniak**
Take 2, let's see if CI is going to tackle this now.

### CODE REVIEW

File: test/e2e/user_workload_monitoring_test.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+			Replicas: proto.Int32(1),
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func assertUserWorkloadMetrics(t *testing.T) {
+	// assert that the previously deployed user application metrics are available in thanos
```
Comment (bwplotka) [inline test/e2e/user_workload_monitoring_test.go]:
```suggestion
	// Assert that the previously deployed user application metrics are available in Thanos.
```

File: test/e2e/user_workload_monitoring_test.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+			if i == 0 {
+				return nil
+			}
+
+			return fmt.Errorf("expected version metric from user application to be equal 0 but got %v", i)
+		},
+	)
+
+	{
+		// assert that the same metric is not scraped by the cluster monitoring stack
```
Comment (bwplotka) [inline test/e2e/user_workload_monitoring_test.go]: ditto

Comment (bwplotka) [reply test/e2e/user_workload_monitoring_test.go]: and below if we want to be matching Go style guide

File: test/e2e/user_workload_monitoring_test.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+	for _, scenario := range []struct {
+		name string
+		f    func(*testing.T)
+	}{
+		{"enable user workload monitoring, assert prometheus rollout", createUserWorkloadAssets(cm)},
+		{"create and assert an user application is deployed", deployUserApplication},
+		{"create prometheus and alertmanager in user namespace", createPrometheusAlertmanagerInUserNamespace},
+		{"assert user workload metrics", assertUserWorkloadMetrics},
+		{"assert prometheus and alertmanager is not deployed in user namespace", assertPrometheusAlertmanagerInUserNamespace},
+		{"assert assets are deleted when user workload monitoring is disabled", assertDeletedUserWorkloadAssets(cm)},
```
Comment (bwplotka) [inline test/e2e/user_workload_monitoring_test.go]: It's kind of weird that `assert` function deletes inside and not only assert, but I it's fine, minor nit.

**Comment by s-urbaniak** [comment]
The e2e test failure is legit, i have a small bug in `createUserWorkloadAssets`. We can only invoke `WaitForPrometheus` and `WaitForStatefulsetRollout` if we know that the statefulset objects have been created in the first place, otherwise the poll will fail immediately.

**Pull request merged**